lib/protos.js

```diff
@@ -113,20 +119,38 @@ exports.map = function(events, event){
   var normalizedEvent = normalize(event);
 
   return foldl(function(matchingEvents, val, key, events) {
-    // If true, this is a `mixed` value, which is structured like so:
+    // There are three types of fields in metadata that can be mapped:
+    // 1) 'array' — array of strings
+    // 3) 'mixed' — array of objects
+    // 2) 'map' - object
+
+    // Mixed values are structured like so:
     //     { key: 'testEvent', value: { event: 'testEvent', someValue: 'xyz' } }
-    // We need to extract the key, which we use to match against
-    // `normalizedEvent`, and return `value` as part of `matchingEvents` if that
-    // match succeds.
+    // In their case, we need to match the normalized `key` field against
+    // `normalizedEvent`, and return whatever is in the `value` field (likely an object) as part
+    // of `matchingEvents` if that match succeds.
+
     if (type(events) === 'array') {
-      // If there's no key attached to this event mapping (unusual), skip this
-      // item.
-      if (!val.key) return matchingEvents;
-      // Extract the key and value from the `mixed` object.
-      key = val.key;
-      val = val.value;
+      switch (type(val)) {
```

Comment:
Per usual, just following precedent: https://github.com/segmentio/analytics.js-integration/blob/master/lib/protos.js#L244-L266

I'm all for clarity, but if we are going to start imposing rules going forward we **need** a style guide so that we have an opportunity to catch *ourselves* before asking for review.